### ISSUE TITLE
Running C++/CX UWP unit test project via vstest.console.exe hangs

### ISSUE THREAD

**Issue by shawnhar** [Microsoft/vstest#1534]
- Clone https://mscodehub.visualstudio.com/_git/Win2D
- Using Visual Studio 15.6.4
- Building and running tests inside the IDE works fine
- But if I run tests from commandline via vstestconsole.exe, the winrt.test.external.uap test run hangs
- Easiest way to repro this is "msbuild win2d.proj /p:BuildPlatforms=Win32 /p:BuildConfigurations=Debug"
- vstest.console.exe is able to launch the test application from its .appx (app window appears) but then hangs – no tests actually run and no log output.  I have to kill the process to recover.
- After attempting to run from vstest.console.exe, any later attempts to run the same tests inside Visual Studio fail with:
```
    [4/4/2018 3:46:31 PM Error] Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Failed to initialize client proxy: could not connect to test process.
       at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources, CancellationToken cancellationToken)
       at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.StartTestRun(TestRunCriteria testRunCriteria, ITestRunEventsHandler eventHandler)
```
- I can fix this situation by uninstalling the test.external application from Windows start menu, after which test runs inside Visual Studio function correctly again

**Comment by jayaranigarg** [comment]
@shawnhar : Thank you for reaching out to us. The support for UWP C++ tests running via appx was not rolled out in VS 15.6.4. This will only be available in latest 15.7 Preview.

As a workaround (for 15.6.4), you can run tests in CLI using appxrecipe(instead of appx). First, uninstall  test.external application from Windows start menu and then try running your tests from CLI like
>vstest.console winrt.test.external.uap.build.appxrecipe /framework:FrameworkUap10

Let us know if this works for you!

**Comment by fguinan** [comment]
I'm having the same issue and don't understand the workaround. What appxrecipe file do I point vstest.console to? When pointing at the appxrecipe file generated in the build output, I get the following message:

_"Test container 'C:\<path-to-build-output>\UnitTestApp2.exe' skipped. The tests in this test container can be executed only in windows store app unit test execution mode
No test is available in C:\<path-to-build-output>\UnitTestApp2.exe. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again."_

This functionality worked fine in v15.5.x, so it is a feature that has already been "rolled out." This new bug is currently blocking our upgrade to the latest version.

Here are similar steps to get to the same issue, using v15.6.6:

1) Launch Visual Studio and create new project/solution, using template "Visual C++ > Windows Universal > Unit Test App".
2) Build Solution and run tests thru Test Explorer. Everything builds and tests fine.
3) Install solution's newly created dev certificate as a trusted cert on the test machine.
4) Launch cmd.exe and navigate to the solution's enclosing directory. Run 'msbuild xxx.sln /t:build /p:Configuration=Debug /p:Platform=x64' , substituting xxx with the newly created solution name.
5) While in cmd.exe, run 'vstest.console AppPackages\xxx\xxx_1.0.0.0_x64_Debug_Test\xxx_1.0.0.0_x64_Debug.appx /InIsolation' , substituting xxx with the newly created solution name.

Result: Test application launches, but tests never run. The command prompt is stuck waiting forever after displaying "Starting test execution, please wait..."

**Comment by jayaranigarg** [comment]
@fguinan : Apologies for the confusion. I edited my previous comment to say that appx support is broken in VS15.6 and will be fixed in VS15.7.

Meanwhile, you can use appxrecipe generated in build output to run your tests. You will have to pass 
**/framework:FrameworkUap10** switch as well while using vstest.console to run your tests.

**Comment by jayaranigarg** [comment]
@shawnhar : Any updates on this? Did the workaround work for you?

**Comment by shawnhar** [comment]
I wasn't able to get the workaround working.   Good to hear this will be fixed in the next VS update though!

**Comment by jayaranigarg** [comment]
@shawnhar : Can you post here the results of running his command on your box here:
>vstest.console winrt.test.external.uap.build.appxrecipe **/framework:FrameworkUap10**

Are you blocked on this or does nest release works for you?

**Comment by jayaranigarg** [comment]
@shawnhar : Any updates?

**Comment by shawnhar** [comment]
I haven't had a chance to try this yet, sorry.  The issue is blocking my ability to run tests during automation, so for now I've just turned that off.

**Comment by mayankbansal018** [comment]
@shawnhar , the support for running tests via ".appx" has been restored from 15.7 preview 3, & will be officially available in 15.7 RTM which is around 7th May.

**Comment by mayankbansal018** [comment]
@shawnhar , please reopen, if you still face this after 15.7 upgrade
